### ISSUE TITLE
Can't find path when run on device

### ISSUE THREAD

**Issue by rain9x** [KiranPanesar/DijkstrasDemo#1]
I have use Dijkstras, when i run on simulator then ok. But when i run on my device then can't find path. I store data on database (include table node, links) to find path.

Please help me!

**Comment by KiranPanesar** [comment]
Are you experiencing this issue when using this project?

**Comment by rain9x** [comment]
Yes! When i run on simulator then very ok. But run on my iPhone then not find path and infinite loop crash my phone.

Please help me

**Comment by KiranPanesar** [comment]
Are you able to trace down the source of the infinite loop?

**Comment by rain9x** [comment]
I have log on xcode:
60's neighbour 59 has weight 512 and is 19 away from node
Working at node 60 with weight 512
60's neighbour 59 has weight 512 and is 19 away from node
Working at node 60 with weight 512
60's neighbour 59 has weight 512 and is 19 away from node
.... infinite loop ----

**Comment by rain9x** [comment]
I fixed it :D!

**Comment by KiranPanesar** [comment]
Was it a bug in this framework? If so, can you push up your code? I would like to take a look and see if we can merge it back in :)
